# [Use Tensorflow DNNClassifier estimator to classify MNIST dataset](https://codeburst.io/use-tensorflow-dnnclassifier-estimator-to-classify-mnist-dataset-a7222bf9f940)

原文很老了，跑出很多 deprecated warning. 分析 150 天行情資料需要用到同樣的技術。Put mnist dataset on MNIST_data/ directory under the working directory with the 4 mist .gz files (on my yun.baidu) in it the we don't need to worry about the dataset and the network downloading and the obsoleted warnings

In [1]:
import tensorflow as tf
import numpy as np
from tensorflow.examples.tutorials.mnist import input_data
import peforth

In [ ]:
%%f  Now we redefine the 'unknown' command that was doing nothing
     : unknown ( token -- thing Y|N) // Try to find the unknown token in __main__
       py> getattr(sys.modules['__main__'],pop(),"Ûnknôwn") 
       py> str(tos())=="Ûnknôwn" if drop false else true then ;
         
     \ here after, when FORTH come accross an unknown token, instead of an error 
     \ message, it try to find the token in python __main__ module name space.

     Redefine \ commant to print it 
     code \ print(nexttoken('\n')) end-code 

In [2]:
mnist = input_data.read_data_sets('MNIST_data')

def input(dataset):
    '''
    傳回 mnist.test mnist.train mnist.validation 看哪個 dataset 當中的 (feature,label) 
    '''
    return dataset.images, dataset.labels.astype(np.int32)
    # where dataset is i.e. mnist.test or mnist.train or mnist.validation

'''
mnist :> test.labels[0] type ==> <class 'numpy.uint8'> 要轉成 np.int32
mnist :> test.images.shape   ==> (10000, 784)
mnist :> test.images[0] type ==> <class 'numpy.ndarray'>
mnist :> test.images[0][0] type ==> <class 'numpy.float32'>
'''

Instructions for updating:
Please use alternatives such as official/mnist/dataset.py from tensorflow/models.
Instructions for updating:
Please write your own downloading logic.
Instructions for updating:
Please use urllib or similar directly.
Successfully downloaded train-images-idx3-ubyte.gz 9912422 bytes.
Instructions for updating:
Please use tf.data to implement this functionality.
Extracting MNIST_data\train-images-idx3-ubyte.gz
Successfully downloaded train-labels-idx1-ubyte.gz 28881 bytes.
Instructions for updating:
Please use tf.data to implement this functionality.
Extracting MNIST_data\train-labels-idx1-ubyte.gz
Successfully downloaded t10k-images-idx3-ubyte.gz 1648877 bytes.
Extracting MNIST_data\t10k-images-idx3-ubyte.gz
Successfully downloaded t10k-labels-idx1-ubyte.gz 4542 bytes.
Extracting MNIST_data\t10k-labels-idx1-ubyte.gz
Instructions for updating:
Please use alternatives such as official/mnist/dataset.py from tensorflow/models.


可參考 GitHub\workshops\notebooks\1-mnist-with-keras.ipynb 看新的怎麼做

    WARNING:tensorflow:From <ipython-input-2-f61e226d87df>:1: read_data_sets (from tensorflow.contrib.learn.python.learn.datasets.mnist) is deprecated and will be removed in a future version.
    Instructions for updating:
    Please use alternatives such as official/mnist/dataset.py from tensorflow/models.
    WARNING:tensorflow:From c:\users\hcche\appdata\local\programs\python\python36\lib\site-packages\tensorflow\contrib\learn\python\learn\datasets\mnist.py:260: maybe_download (from tensorflow.contrib.learn.python.learn.datasets.base) is deprecated and will be removed in a future version.
    Instructions for updating:
    Please write your own downloading logic.
    WARNING:tensorflow:From c:\users\hcche\appdata\local\programs\python\python36\lib\site-packages\tensorflow\contrib\learn\python\learn\datasets\base.py:252: _internal_retry.<locals>.wrap.<locals>.wrapped_fn (from tensorflow.contrib.learn.python.learn.datasets.base) is deprecated and will be removed in a future version.
    Instructions for updating:
    Please use urllib or similar directly.
    Successfully downloaded train-images-idx3-ubyte.gz 9912422 bytes.
    WARNING:tensorflow:From c:\users\hcche\appdata\local\programs\python\python36\lib\site-packages\tensorflow\contrib\learn\python\learn\datasets\mnist.py:262: extract_images (from tensorflow.contrib.learn.python.learn.datasets.mnist) is deprecated and will be removed in a future version.
    Instructions for updating:
    Please use tf.data to implement this functionality.
    Extracting MNIST_data\train-images-idx3-ubyte.gz
    Successfully downloaded train-labels-idx1-ubyte.gz 28881 bytes.
    WARNING:tensorflow:From c:\users\hcche\appdata\local\programs\python\python36\lib\site-packages\tensorflow\contrib\learn\python\learn\datasets\mnist.py:267: extract_labels (from tensorflow.contrib.learn.python.learn.datasets.mnist) is deprecated and will be removed in a future version.
    Instructions for updating:
    Please use tf.data to implement this functionality.
    Extracting MNIST_data\train-labels-idx1-ubyte.gz
    Successfully downloaded t10k-images-idx3-ubyte.gz 1648877 bytes.
    Extracting MNIST_data\t10k-images-idx3-ubyte.gz
    Successfully downloaded t10k-labels-idx1-ubyte.gz 4542 bytes.
    Extracting MNIST_data\t10k-labels-idx1-ubyte.gz
    WARNING:tensorflow:From c:\users\hcche\appdata\local\programs\python\python36\lib\site-packages\tensorflow\contrib\learn\python\learn\datasets\mnist.py:290: DataSet.__init__ (from tensorflow.contrib.learn.python.learn.datasets.mnist) is deprecated and will be removed in a future version.
    Instructions for updating:
    Please use alternatives such as official/mnist/dataset.py from tensorflow/models.

In [3]:
# Specify feature
feature_columns = [tf.feature_column.numeric_column("x", shape=[28, 28])]

In [91]:
%%f 查看上面這 feature_columns 啥玩意兒?
tf py> [pop().feature_column.numeric_column("xx",shape=[8,8])] dup . cr \ "xx" 可以亂寫
type . cr 
feature_columns type . cr
\ 這玩意兒應該是 iterator 之類的東西 ==> 錯! 它是個 class
tf :> feature_column.numeric_column("abc",shape=[8,8]) dup ( 留一份 ) type . cr
py> [pop()] . cr \ 把這個 class 放進 [...] 裡去，似乎有點 new 出來的味道，但 "xx" 是亂寫的故不然。

\ 總之，這已經解答了 150 天行情該怎麼表達的問題。答案是:
tf py> [pop().feature_column.numeric_column("day150",shape=[150,5])] dup . cr
\ 比土土的 750 flattend 要有意義。 

[_NumericColumn(key='xx', shape=(8, 8), default_value=None, dtype=tf.float32, normalizer_fn=None)]
"xx" 可以亂寫
<class 'list'>
<class 'list'>
這玩意兒應該是 iterator 之類的東西 ==> 錯! 它是個 class
<class 'tensorflow.python.feature_column.feature_column._NumericColumn'>
[_NumericColumn(key='abc', shape=(8, 8), default_value=None, dtype=tf.float32, normalizer_fn=None)]
把這個 class 放進 [...] 裡去，似乎有點 new 出來的味道，但 "xx" 是亂寫的故不然。
總之，這已經解答了 150 天行情該怎麼表達的問題。答案是:
[_NumericColumn(key='day150', shape=(150, 5), default_value=None, dtype=tf.float32, normalizer_fn=None)]
比土土的 750 flattend 要有意義。 


In [4]:
# Build 2 layer DNN classifier
classifier = tf.estimator.DNNClassifier(
    feature_columns=feature_columns,
    hidden_units=[256, 32],
    optimizer=tf.train.AdamOptimizer(1e-4),
    n_classes=10,
    dropout=0.1,
    model_dir="./tmp/mnist_model"
    )

INFO:tensorflow:Using default config.
INFO:tensorflow:Using config: {'_model_dir': './tmp/mnist_model', '_tf_random_seed': None, '_save_summary_steps': 100, '_save_checkpoints_steps': None, '_save_checkpoints_secs': 600, '_session_config': None, '_keep_checkpoint_max': 5, '_keep_checkpoint_every_n_hours': 10000, '_log_step_count_steps': 100, '_train_distribute': None, '_service': None, '_cluster_spec': <tensorflow.python.training.server_lib.ClusterSpec object at 0x00000203431180F0>, '_task_type': 'worker', '_task_id': 0, '_global_id_in_cluster': 0, '_master': '', '_evaluation_master': '', '_is_chief': True, '_num_ps_replicas': 0, '_num_worker_replicas': 1}


In [93]:
%f \ 看清楚等下要用的 x y 啥東西
x = {"x": input(mnist.train)[0]}
y = input(mnist.train)[1]
%f x :> ['x'].shape tib.
%f y :> shape tib.
%f x :> ['x'][:4] tib.

看清楚等下要用的 x y 啥東西
x :> ['x'].shape tib. \ ==> (55000, 784) (<class 'tuple'>)
y :> shape tib. \ ==> (55000,) (<class 'tuple'>)
x :> ['x'][:4] tib. \ ==> [[ 0.  0.  0. ...,  0.  0.  0.]
 [ 0.  0.  0. ...,  0.  0.  0.]
 [ 0.  0.  0. ...,  0.  0.  0.]
 [ 0.  0.  0. ...,  0.  0.  0.]] (<class 'numpy.ndarray'>)


In [5]:
# Define the training inputs
train_input_fn = tf.estimator.inputs.numpy_input_fn(
    x={"x": input(mnist.train)[0]},
    y=input(mnist.train)[1],
    num_epochs=None,
    batch_size=50,
    shuffle=True
    )

In [6]:
classifier.train(input_fn=train_input_fn, steps=100000)

INFO:tensorflow:Calling model_fn.
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Create CheckpointSaverHook.
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Saving checkpoints for 1 into ./tmp/mnist_model\model.ckpt.
INFO:tensorflow:loss = 114.947, step = 1
INFO:tensorflow:global_step/sec: 294.284
INFO:tensorflow:loss = 77.4226, step = 101 (0.344 sec)
INFO:tensorflow:global_step/sec: 346.219
INFO:tensorflow:loss = 56.7334, step = 201 (0.287 sec)
INFO:tensorflow:global_step/sec: 369.217
INFO:tensorflow:loss = 38.599, step = 301 (0.271 sec)
INFO:tensorflow:global_step/sec: 371.96
INFO:tensorflow:loss = 33.9613, step = 401 (0.269 sec)
INFO:tensorflow:global_step/sec: 363.845
INFO:tensorflow:loss = 21.9289, step = 501 (0.276 sec)
INFO:tensorflow:global_step/sec: 362.525
INFO:tensorflow:loss = 21.7903, step = 601 (0.276 sec)
INFO:tensorflow:global_step/sec: 365.173
INFO:tensorflow:loss = 14.4352,

INFO:tensorflow:loss = 7.84252, step = 8201 (0.297 sec)
INFO:tensorflow:global_step/sec: 357.346
INFO:tensorflow:loss = 6.98077, step = 8301 (0.280 sec)
INFO:tensorflow:global_step/sec: 363.845
INFO:tensorflow:loss = 2.02472, step = 8401 (0.274 sec)
INFO:tensorflow:global_step/sec: 371.96
INFO:tensorflow:loss = 2.52588, step = 8501 (0.269 sec)
INFO:tensorflow:global_step/sec: 366.511
INFO:tensorflow:loss = 2.31865, step = 8601 (0.274 sec)
INFO:tensorflow:global_step/sec: 354.812
INFO:tensorflow:loss = 12.1357, step = 8701 (0.282 sec)
INFO:tensorflow:global_step/sec: 367.857
INFO:tensorflow:loss = 8.47307, step = 8801 (0.272 sec)
INFO:tensorflow:global_step/sec: 251.711
INFO:tensorflow:loss = 3.9724, step = 8901 (0.396 sec)
INFO:tensorflow:global_step/sec: 365.173
INFO:tensorflow:loss = 10.9184, step = 9001 (0.275 sec)
INFO:tensorflow:global_step/sec: 365.171
INFO:tensorflow:loss = 4.6987, step = 9101 (0.274 sec)
INFO:tensorflow:global_step/sec: 367.859
INFO:tensorflow:loss = 8.47744, s

INFO:tensorflow:loss = 1.91765, step = 16601 (0.309 sec)
INFO:tensorflow:global_step/sec: 324.861
INFO:tensorflow:loss = 1.42157, step = 16701 (0.308 sec)
INFO:tensorflow:global_step/sec: 347.424
INFO:tensorflow:loss = 11.7321, step = 16801 (0.289 sec)
INFO:tensorflow:global_step/sec: 335.762
INFO:tensorflow:loss = 2.03756, step = 16901 (0.298 sec)
INFO:tensorflow:global_step/sec: 328.056
INFO:tensorflow:loss = 1.88494, step = 17001 (0.305 sec)
INFO:tensorflow:global_step/sec: 303.203
INFO:tensorflow:loss = 8.34148, step = 17101 (0.330 sec)
INFO:tensorflow:global_step/sec: 329.136
INFO:tensorflow:loss = 3.16712, step = 17201 (0.304 sec)
INFO:tensorflow:global_step/sec: 334.64
INFO:tensorflow:loss = 7.71781, step = 17301 (0.301 sec)
INFO:tensorflow:global_step/sec: 330.222
INFO:tensorflow:loss = 1.97458, step = 17401 (0.301 sec)
INFO:tensorflow:global_step/sec: 343.84
INFO:tensorflow:loss = 3.94022, step = 17501 (0.290 sec)
INFO:tensorflow:global_step/sec: 338.03
INFO:tensorflow:loss = 

INFO:tensorflow:loss = 0.797982, step = 25001 (0.308 sec)
INFO:tensorflow:global_step/sec: 320.697
INFO:tensorflow:loss = 0.583521, step = 25101 (0.313 sec)
INFO:tensorflow:global_step/sec: 338.033
INFO:tensorflow:loss = 1.49442, step = 25201 (0.296 sec)
INFO:tensorflow:global_step/sec: 338.03
INFO:tensorflow:loss = 2.37978, step = 25301 (0.296 sec)
INFO:tensorflow:global_step/sec: 339.178
INFO:tensorflow:loss = 0.659714, step = 25401 (0.295 sec)
INFO:tensorflow:global_step/sec: 331.314
INFO:tensorflow:loss = 1.10717, step = 25501 (0.302 sec)
INFO:tensorflow:global_step/sec: 339.178
INFO:tensorflow:loss = 8.53239, step = 25601 (0.295 sec)
INFO:tensorflow:global_step/sec: 329.131
INFO:tensorflow:loss = 0.651182, step = 25701 (0.304 sec)
INFO:tensorflow:global_step/sec: 334.642
INFO:tensorflow:loss = 3.08507, step = 25801 (0.299 sec)
INFO:tensorflow:global_step/sec: 332.418
INFO:tensorflow:loss = 5.58207, step = 25901 (0.300 sec)
INFO:tensorflow:global_step/sec: 336.893
INFO:tensorflow:l

INFO:tensorflow:global_step/sec: 333.524
INFO:tensorflow:loss = 0.345862, step = 33401 (0.298 sec)
INFO:tensorflow:global_step/sec: 348.631
INFO:tensorflow:loss = 0.674459, step = 33501 (0.288 sec)
INFO:tensorflow:global_step/sec: 326.984
INFO:tensorflow:loss = 0.747047, step = 33601 (0.308 sec)
INFO:tensorflow:global_step/sec: 304.123
INFO:tensorflow:loss = 1.22866, step = 33701 (0.328 sec)
INFO:tensorflow:global_step/sec: 335.767
INFO:tensorflow:loss = 3.40977, step = 33801 (0.297 sec)
INFO:tensorflow:global_step/sec: 351.077
INFO:tensorflow:loss = 2.30065, step = 33901 (0.285 sec)
INFO:tensorflow:global_step/sec: 330.222
INFO:tensorflow:loss = 1.22759, step = 34001 (0.304 sec)
INFO:tensorflow:global_step/sec: 314.645
INFO:tensorflow:loss = 2.96396, step = 34101 (0.317 sec)
INFO:tensorflow:global_step/sec: 332.415
INFO:tensorflow:loss = 0.3737, step = 34201 (0.301 sec)
INFO:tensorflow:global_step/sec: 338.033
INFO:tensorflow:loss = 0.623019, step = 34301 (0.296 sec)
INFO:tensorflow:g

INFO:tensorflow:loss = 1.62659, step = 41701 (0.320 sec)
INFO:tensorflow:global_step/sec: 318.653
INFO:tensorflow:loss = 0.443529, step = 41801 (0.314 sec)
INFO:tensorflow:global_step/sec: 321.729
INFO:tensorflow:loss = 0.165862, step = 41901 (0.311 sec)
INFO:tensorflow:global_step/sec: 323.808
INFO:tensorflow:loss = 1.33656, step = 42001 (0.309 sec)
INFO:tensorflow:global_step/sec: 322.767
INFO:tensorflow:loss = 0.624092, step = 42101 (0.310 sec)
INFO:tensorflow:global_step/sec: 330.221
INFO:tensorflow:loss = 0.201291, step = 42201 (0.302 sec)
INFO:tensorflow:global_step/sec: 330.224
INFO:tensorflow:loss = 0.487753, step = 42301 (0.304 sec)
INFO:tensorflow:global_step/sec: 312.678
INFO:tensorflow:loss = 3.1607, step = 42401 (0.320 sec)
INFO:tensorflow:global_step/sec: 310.735
INFO:tensorflow:loss = 0.692839, step = 42501 (0.323 sec)
INFO:tensorflow:global_step/sec: 318.656
INFO:tensorflow:loss = 0.644669, step = 42601 (0.312 sec)
INFO:tensorflow:global_step/sec: 316.636
INFO:tensorflo

INFO:tensorflow:global_step/sec: 316.637
INFO:tensorflow:loss = 0.180131, step = 50101 (0.315 sec)
INFO:tensorflow:global_step/sec: 346.219
INFO:tensorflow:loss = 0.325396, step = 50201 (0.290 sec)
INFO:tensorflow:global_step/sec: 345.025
INFO:tensorflow:loss = 0.118854, step = 50301 (0.290 sec)
INFO:tensorflow:global_step/sec: 342.663
INFO:tensorflow:loss = 0.487472, step = 50401 (0.292 sec)
INFO:tensorflow:global_step/sec: 342.661
INFO:tensorflow:loss = 0.0433097, step = 50501 (0.291 sec)
INFO:tensorflow:global_step/sec: 349.851
INFO:tensorflow:loss = 0.308025, step = 50601 (0.286 sec)
INFO:tensorflow:global_step/sec: 353.558
INFO:tensorflow:loss = 0.4689, step = 50701 (0.285 sec)
INFO:tensorflow:global_step/sec: 335.763
INFO:tensorflow:loss = 0.230209, step = 50801 (0.296 sec)
INFO:tensorflow:global_step/sec: 352.315
INFO:tensorflow:loss = 0.280561, step = 50901 (0.285 sec)
INFO:tensorflow:global_step/sec: 336.893
INFO:tensorflow:loss = 0.121718, step = 51001 (0.298 sec)
INFO:tensor

INFO:tensorflow:global_step/sec: 332.415
INFO:tensorflow:loss = 0.208349, step = 58401 (0.299 sec)
INFO:tensorflow:global_step/sec: 352.314
INFO:tensorflow:loss = 0.392357, step = 58501 (0.284 sec)
INFO:tensorflow:global_step/sec: 343.838
INFO:tensorflow:loss = 0.0477803, step = 58601 (0.291 sec)
INFO:tensorflow:global_step/sec: 347.423
INFO:tensorflow:loss = 0.682486, step = 58701 (0.289 sec)
INFO:tensorflow:global_step/sec: 347.419
INFO:tensorflow:loss = 0.120645, step = 58801 (0.288 sec)
INFO:tensorflow:global_step/sec: 348.631
INFO:tensorflow:loss = 0.426344, step = 58901 (0.286 sec)
INFO:tensorflow:global_step/sec: 346.222
INFO:tensorflow:loss = 0.37495, step = 59001 (0.289 sec)
INFO:tensorflow:global_step/sec: 351.078
INFO:tensorflow:loss = 0.762002, step = 59101 (0.286 sec)
INFO:tensorflow:global_step/sec: 346.22
INFO:tensorflow:loss = 0.0731512, step = 59201 (0.289 sec)
INFO:tensorflow:global_step/sec: 343.838
INFO:tensorflow:loss = 0.343858, step = 59301 (0.291 sec)
INFO:tenso

INFO:tensorflow:global_step/sec: 324.862
INFO:tensorflow:loss = 3.30212, step = 66701 (0.309 sec)
INFO:tensorflow:global_step/sec: 348.631
INFO:tensorflow:loss = 0.325082, step = 66801 (0.288 sec)
INFO:tensorflow:global_step/sec: 340.33
INFO:tensorflow:loss = 0.263977, step = 66901 (0.292 sec)
INFO:tensorflow:global_step/sec: 351.079
INFO:tensorflow:loss = 0.112859, step = 67001 (0.285 sec)
INFO:tensorflow:global_step/sec: 349.85
INFO:tensorflow:loss = 0.0771247, step = 67101 (0.287 sec)
INFO:tensorflow:global_step/sec: 351.077
INFO:tensorflow:loss = 0.193813, step = 67201 (0.285 sec)
INFO:tensorflow:global_step/sec: 349.851
INFO:tensorflow:loss = 0.202688, step = 67301 (0.286 sec)
INFO:tensorflow:global_step/sec: 342.662
INFO:tensorflow:loss = 0.372941, step = 67401 (0.291 sec)
INFO:tensorflow:global_step/sec: 345.023
INFO:tensorflow:loss = 0.800723, step = 67501 (0.291 sec)
INFO:tensorflow:global_step/sec: 341.494
INFO:tensorflow:loss = 0.187482, step = 67601 (0.293 sec)
INFO:tensorf

INFO:tensorflow:global_step/sec: 321.727
INFO:tensorflow:loss = 0.0780363, step = 75001 (0.312 sec)
INFO:tensorflow:global_step/sec: 335.762
INFO:tensorflow:loss = 0.0629612, step = 75101 (0.297 sec)
INFO:tensorflow:global_step/sec: 347.419
INFO:tensorflow:loss = 0.240897, step = 75201 (0.287 sec)
INFO:tensorflow:global_step/sec: 343.841
INFO:tensorflow:loss = 0.272562, step = 75301 (0.291 sec)
INFO:tensorflow:global_step/sec: 341.493
INFO:tensorflow:loss = 0.0266555, step = 75401 (0.294 sec)
INFO:tensorflow:global_step/sec: 332.416
INFO:tensorflow:loss = 0.00467187, step = 75501 (0.301 sec)
INFO:tensorflow:global_step/sec: 342.661
INFO:tensorflow:loss = 0.110861, step = 75601 (0.291 sec)
INFO:tensorflow:global_step/sec: 352.315
INFO:tensorflow:loss = 0.0666749, step = 75701 (0.285 sec)
INFO:tensorflow:global_step/sec: 339.169
INFO:tensorflow:loss = 0.281567, step = 75801 (0.296 sec)
INFO:tensorflow:global_step/sec: 338.039
INFO:tensorflow:loss = 0.46787, step = 75901 (0.295 sec)
INFO:

INFO:tensorflow:global_step/sec: 326.985
INFO:tensorflow:loss = 0.0636878, step = 83301 (0.306 sec)
INFO:tensorflow:global_step/sec: 345.024
INFO:tensorflow:loss = 0.0759743, step = 83401 (0.292 sec)
INFO:tensorflow:global_step/sec: 338.032
INFO:tensorflow:loss = 0.0348904, step = 83501 (0.294 sec)
INFO:tensorflow:global_step/sec: 342.661
INFO:tensorflow:loss = 0.075032, step = 83601 (0.293 sec)
INFO:tensorflow:global_step/sec: 338.031
INFO:tensorflow:loss = 0.159042, step = 83701 (0.295 sec)
INFO:tensorflow:global_step/sec: 340.332
INFO:tensorflow:loss = 0.0148849, step = 83801 (0.295 sec)
INFO:tensorflow:global_step/sec: 335.762
INFO:tensorflow:loss = 0.058094, step = 83901 (0.298 sec)
INFO:tensorflow:global_step/sec: 341.492
INFO:tensorflow:loss = 0.0145958, step = 84001 (0.293 sec)
INFO:tensorflow:global_step/sec: 340.33
INFO:tensorflow:loss = 0.162665, step = 84101 (0.297 sec)
INFO:tensorflow:global_step/sec: 332.418
INFO:tensorflow:loss = 0.0754027, step = 84201 (0.298 sec)
INFO:

INFO:tensorflow:global_step/sec: 319.672
INFO:tensorflow:loss = 0.122287, step = 91601 (0.313 sec)
INFO:tensorflow:global_step/sec: 351.08
INFO:tensorflow:loss = 0.0512695, step = 91701 (0.285 sec)
INFO:tensorflow:global_step/sec: 342.66
INFO:tensorflow:loss = 0.0108757, step = 91801 (0.292 sec)
INFO:tensorflow:global_step/sec: 348.633
INFO:tensorflow:loss = 0.115683, step = 91901 (0.287 sec)
INFO:tensorflow:global_step/sec: 335.764
INFO:tensorflow:loss = 0.397654, step = 92001 (0.298 sec)
INFO:tensorflow:global_step/sec: 346.218
INFO:tensorflow:loss = 0.055269, step = 92101 (0.289 sec)
INFO:tensorflow:global_step/sec: 343.839
INFO:tensorflow:loss = 0.0394457, step = 92201 (0.290 sec)
INFO:tensorflow:global_step/sec: 342.663
INFO:tensorflow:loss = 0.0403066, step = 92301 (0.293 sec)
INFO:tensorflow:global_step/sec: 340.33
INFO:tensorflow:loss = 0.0396517, step = 92401 (0.294 sec)
INFO:tensorflow:global_step/sec: 346.219
INFO:tensorflow:loss = 0.0519229, step = 92501 (0.289 sec)
INFO:te

INFO:tensorflow:global_step/sec: 262.618
INFO:tensorflow:loss = 1.3603, step = 99901 (0.381 sec)
INFO:tensorflow:Saving checkpoints for 100000 into ./tmp/mnist_model\model.ckpt.
INFO:tensorflow:Loss for final step: 0.0563969.


In [94]:
%f \ 看清楚等下要用的 x y 啥東西
x = {"x": input(mnist.test)[0]}
y = input(mnist.test)[1]
%f x :> ['x'].shape tib.
%f y :> shape tib.
%f x :> ['x'][:4] tib.

看清楚等下要用的 x y 啥東西
x :> ['x'].shape tib. \ ==> (10000, 784) (<class 'tuple'>)
y :> shape tib. \ ==> (10000,) (<class 'tuple'>)
x :> ['x'][:4] tib. \ ==> [[ 0.  0.  0. ...,  0.  0.  0.]
 [ 0.  0.  0. ...,  0.  0.  0.]
 [ 0.  0.  0. ...,  0.  0.  0.]
 [ 0.  0.  0. ...,  0.  0.  0.]] (<class 'numpy.ndarray'>)


In [7]:
# Define the test inputs
test_input_fn = tf.estimator.inputs.numpy_input_fn(
    x={"x": input(mnist.test)[0]},
    y=input(mnist.test)[1],
    num_epochs=1,
    shuffle=False
    )

In [8]:
# Evaluate accuracy
accuracy_score = classifier.evaluate(input_fn=test_input_fn)["accuracy"]
print("\nTest Accuracy: {0:f}%\n".format(accuracy_score*100))

# End of "mnist_estimator.py"


INFO:tensorflow:Calling model_fn.
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Starting evaluation at 2018-05-23-03:40:55
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from ./tmp/mnist_model\model.ckpt-100000
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Finished evaluation at 2018-05-23-03:40:55
INFO:tensorflow:Saving dict for global step 100000: accuracy = 0.9816, average_loss = 0.0811812, global_step = 100000, loss = 10.2761

Test Accuracy: 98.159999%



# --- The End ---